### ISSUE TITLE
Two errors after updating to HA core 2021.12.0

### ISSUE THREAD

**Issue by Flash3dee** [DCSBL/ha-homewizard-energy#68]
After upgrading to Home Assistant 2021.12.0 I'm getting two errors in the HA log. 

*A clear and concise description of what the bug is.*

Made this report from my phone, so excuses if not filled in properly. 
Was upgrading from latest November Core build. Using 0.12.0 version of the HW integration. 

**Environment (please complete the following information)**

 - Integration version:
 - Home Assistant version:
 - Python version (if known):

**To Reproduce**

Eg. Steps to reproduce the behavior:

**Expected behavior**

*A clear and concise description of what you expected to happen.*

**Screenshots**

*If applicable, add screenshots to help explain your problem.*

**Log Output**
Logger: homeassistant.helpers.frame
Source: helpers/frame.py:103
First occurred: 11:00:55 (1 occurrences)
Last logged: 11:00:55

Detected integration that accessed discovery_info['0'] instead of discovery_info.0; this will fail in version 2022.6. Please report issue to the custom component author for homewizard_energy using this method at custom_components/homewizard_energy/config_flow.py, line 63: "host" not in discovery_info

And

Logger: homeassistant
Source: components/zeroconf/__init__.py:126
First occurred: 11:00:55 (7 occurrences)
Last logged: 11:23:51

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/discovery_flow.py", line 74, in _async_process_pending_flows
    await gather_with_concurrency(
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 170, in gather_with_concurrency
    return await gather(
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 168, in sem_task
    return await task
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 203, in async_init
    flow, result = await task
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 230, in _async_init
    result = await self._async_handle_step(flow, flow.init_step, data, init_done)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 325, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/homewizard_energy/config_flow.py", line 63, in async_step_zeroconf
    "host" not in discovery_info
  File "/usr/src/homeassistant/homeassistant/components/zeroconf/__init__.py", line 126, in __getitem__
    return getattr(self, name)
TypeError: getattr(): attribute name must be string

*If applicable*
To get more output, enable `debug` output
```yaml
# configuration.yaml
logger:
  default: info
  logs:
    custom_components.homewizard_energy: debug
```

**Additional context**

*Add any other context about the problem here.*

**Comment by DCSBL** [comment]
Thanks!
I've updated what was asked with [0.12.1-beta-1](https://github.com/DCSBL/ha-homewizard-energy/releases/tag/0.12.1-beta-1). It is available on HACS if you select 'Show beta versions'. Can you try to see if this fixes your issue?

**Comment by Flash3dee** [comment]
Not home now so will try this evening. Thanks for the swift fix, will it fix both of the errors?

**Comment by Flash3dee** [comment]
Just tried 0.12.1 beta 1 , all errors fixed! TY.

**Comment by DCSBL** [comment]
Awesome. Released 0.12.1 with this fix. Thank you for reporting 👍
